src/reducers/icons.js

```diff
@@ -0,0 +1,21 @@
+import Immutable from 'immutable'
+
+/**
+ * The Icons reducer
+ *
+ * @param state
+ * @param action
+ * @returns {*}
+ */
+function icons (state, action) {
+  state = state || Immutable.fromJS({})
```

Comment:
please prefer `Immutable.Map` to `Immutable.fromJS({})`. Overall I would suggest restructuring this code a bit:
```javascript
import { Map } From 'immutable'

function icons (state = Map(), action) {
    // ... rest of the code
}
```